### ISSUE TITLE
ios_config 'save' option was removed in 2.4 still has deprecation as only a warning

### ISSUE THREAD

**Issue by bdowling** [ansible/ansible#32980]
##### ISSUE TYPE

 - Bug Report

##### COMPONENT NAME

ios_config

##### ANSIBLE VERSION

```
ansible 2.4.0.0
  config file = /data/conf/netdev/plays/ansible.cfg
  configured module search path = [u'/data/conf/netdev/extensions/modules']
  ansible python module location = /usr/lib/python2.7/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 2.7.5 (default, Aug  4 2017, 00:39:18) [GCC 4.8.5 20150623 (Red Hat 4.8.5-16)]
```

##### SUMMARY

Ansible 2.4 has a deprecation warning for the 'save' option being removed in 2.4.  Since it has been removed, this should likely be a fatal error using 'save' in the play.

##### STEPS TO REPRODUCE
Run the below play

```yaml
  - name: save test
    ios_config:
      authorize: yes
      save: yes
```

##### EXPECTED RESULTS
Fatal error message on invalid ios_config save option.

##### ACTUAL RESULTS
```
[DEPRECATION WARNING]: Param 'save' is deprecated. See the module docs for more information. This feature will be removed in version 2.4. Deprecation warnings  can be disabled by setting deprecation_warnings=False in ansible.cfg.

```
##### Other References
Fwiw, this has been mentioned in another ticket https://github.com/ansible/ansible/pull/32744#issuecomment-344660626

@gundalow also asked me to open a issue to see if this could slide into the 2.4.2 release.

Hoping the the save_now prompting fix also is included #32619

**Comment by ansibot** [comment]
cc @Qalthos @ganeshrn @kedarX @privateip @rcarrillocruz @trishnaguha
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by bdowling** [comment]
@termlen0 just fyi regarding your comment I referenced above.  I reread it and noted one thing.  You mentioned that save works in 2.4.  **It does not**, it is a NOOP, at least in the 2.4.0 code in the centos RPM ansible-2.4.0.0-5.el7.noarch

It looks to me as if 'save' silently stopped working as soon as 'save_when' was added, but I could be mistaken.
https://github.com/ansible/ansible/commit/0b6f0e6c0d95e271cf4173b28ece4a5e58902f9c#diff-0ca42d7f4ac4dd7fb03025f0189851f6L332
